puppet/modules/jenkins_job_builder/files/theforeman.org/pipelines/test/testKatello.groovy

```diff
@@ -0,0 +1,141 @@
+def ruby = '2.4'
+def katello_versions = [
+    'master': 'develop',
+    'KATELLO-3.6': '1.17-stable',
+    'KATELLO-3.5': '1.16-stable'
+]
+
+pipeline {
+    agent { label 'fast' }
+
+    stages {
+        stage('Setup Git Repos') {
+            steps {
+               checkout changelog: true, poll: false, scm: [
+                   $class: 'GitSCM',
+                   branches: [[name: '${ghprbActualCommit}']],
+                   doGenerateSubmoduleConfigurations: false,
+                   extensions: [[$class: 'PreBuildMerge', options: [fastForwardMode: 'FF', mergeRemote: 'origin', mergeStrategy: 'default', mergeTarget: '${ghprbTargetBranch}']]],
+                   userRemoteConfigs: [
+                       [refspec: '+refs/pull/${ghprbPullId}/*:refs/remotes/origin/pr/${ghprbPullId}/*', url: 'https://github.com/katello/katello']
+                   ]
+               ]
+
+                dir('foreman') {
+                    git url: "https://github.com/theforeman/foreman", branch: katello_versions[ghprbTargetBranch]
+                }
+            }
+        }
+        stage("Setup RVM") {
+            steps {
+
+                configureRVM(ruby)
+
+            }
+        }
+        stage('Configure Environment') {
+            steps {
+
+                dir('foreman') {
+                    addSettings([
+                        organizations: true,
+                        locations: true
+                    ])
+                    addGem()
+                    databaseFile(gemset())
+                }
+
+            }
+        }
+        stage('Configure Database') {
+            steps {
+
+                dir('foreman') {
+                    configureDatabase(ruby)
+                }
+
+            }
+        }
+        stage('Run Tests') {
+            parallel {
```

Comment:
@evgeni Thanks for your question in this area it helped me fine this syntax which allows for https://github.com/theforeman/foreman-infra/pull/321/files#diff-912a62e12e77edab1326c08754e1b7e9R77